src/main/java/com/google/cloud/teleport/templates/common/SplunkConverters.java

```diff
@@ -233,6 +238,11 @@ public void processElement(ProcessContext context) {
                             if (!event.isEmpty()) {
                               builder.withEvent(event);
                             }
+                            
+                            String fields = metadata.optString(HEC_FIELDS_KEY);
+                            if (!fields.isEmpty()) {
+                              builder.withFields(GSON.fromJson(fields, JsonObject.class));
```

Comment:
What happens if fields is not in the proper format (JSON object)?
Note empty object {} is also valid.

I think we have 3 options:
A. Send event with metadata parsing (time, etc.) except for fields
B. Send event as is (without any metadata parsing)
C. Drop event in DL topic

I would vote for A to avoid data loss (or high recovery burden due to failed optional fields extraction), while also logging warnings for visibility/troubleshooting. 
In all cases, I think we should log a warning, and so probably need a separate try/catch block for this.

Reply:
Per our discussions, I've added more tests for corner-cases and logged exceptions as warnings while parsing `fields` and `timestamp`.

Reply:
Option A sounds good to me, too. 👍